libraries/ms-common/src/objpool.c

```diff
@@ -1,59 +1,66 @@
-// The object pool uses a flag and index to verify a given node and determine its state.
-// We currently use a bitfield to maximize the index limit.
+// We use a bitset to represent free nodes
 #include <stdbool.h>
 #include <string.h>
 
 #include "objpool.h"
 #include "status.h"
 
 #define OBJPOOL_GET(pool, index) \
-  ((ObjectMarker *)((uintptr_t)(pool)->nodes + ((index) * (pool)->node_size)))
+  ((void *)((uintptr_t)(pool)->nodes + ((index) * (pool)->node_size)))
+
+// Check in range and alignment
+#define OBJPOOL_NODE_INVALID(pool, node) \
+  ((pool)->nodes > (node) || \
+   ((pool)->nodes + (pool)->num_nodes * (pool)->node_size) <= (node) || \
```

Comment:
The lint failed because of this, but I'm aligning these lines to the opening brace. Thoughts?